pipelines/DarkEnergyCamera/ApPipe.yaml

```diff
@@ -0,0 +1,35 @@
+description: End to end AP pipeline specialized for DECam
+# (1) Run $AP_PIPE_DIR/pipelines/DarkEnergyCamera/RunIsrForCrosstalkSources.yaml
+# (2) Execute `make_apdb.py` with the same configs used here for diaPipe, e.g.,
+#     make_apdb.py -c isolation_level=READ_UNCOMMITTED
+#                  -c db_url="sqlite:////project/user/association.db"
+# (3) Run this pipeline
+
+instrument: lsst.obs.decam.DarkEnergyCamera
+imports:
+  - location: $AP_PIPE_DIR/pipelines/ApPipe.yaml
+    exclude:
+      - isr
+  - location: $AP_PIPE_DIR/pipelines/DarkEnergyCamera/RunIsrWithCrosstalk.yaml
+tasks:
+    characterizeImage:
+        class: lsst.pipe.tasks.characterizeImage.CharacterizeImageTask
+        config:
+          refObjLoader.ref_dataset_name: 'panstarrs'
```

Comment:
Why are these refcat overrides needed for DECam, specifically? It seems to me like it should depend on what repo you're processing.

Reply:
Both DECam ISR pipelines override `outputExposure` to `"postISRCCD"`. Does there need to be a similar input override?

Reply:
There isn't anything DECam in `/repo/main`, so I can't say for sure yet. This is what I had to do to get it to work with my gen3 repo in `/project/mrawls/hits2015-3`. How does `ap_verify` handle gen3 refcats with the HiTS datasets?

Good catch re. the `outputExposure` setting, that appears to be an extraneously specified default, so I'll remove it.

Reply:
I answered the first part of my own question by looking at the chained list of `/repo/main` collections. The answer is yes, `HSC/defaults` includes `HSC/raw/all`, `HSC/calib`, `refcats`, and `skymaps`.

Reply:
For `ap_verify`, the overrides are applied as part of dataset-specific configs. They're not included in the pipeline.

Reply:
Hmm, I guess I'm frustrated there is no good place to set sensible defaults to prepare for a reality with no ap_verify-style datasets.

I took a look at `ap_verify_ci_hits2015` and I do see the appropriate settings in `config/calibrate.py` to `panstarrs` and `gaia`. It appears to use the obs_decam default for characterizeImage, which is `ps1_pv3_3pi_20170110`, but only for Gen2 🤯 I don't know how the characterizeImage ref_dataset_name is set in Gen3.

Bottom line, I would like users to be able to run this pipeline out of the box with whatever the refcat names in `/repo/main` for panstarrs and gaia wind up being, which will likely be `panstarrs` and `gaia`, for photometry and astrometry respectively. Plus the mappings to `phot_g_mean` are basically always correct. Suggestions?

Reply:
The `characterizeImage` thing sounds like a hidden bug. Maybe it's not being used?

As for what to do, the `ps1_pv3_3pi_20170110` defaults were motivated by those datasets being set up on `lsst-dev`, but I'm guessing we can't change those while Gen 2 is still in use. I think the RFC-775-ish solution is to create a pipeline specialized for `/repo/main`?

Though now I'm wondering why you didn't have to do this for HSC...

Reply:
I think there are good defaults for HSC refcats assuming everything is panstarrs somewhere (in obs_subaru) that obs_decam lacks. When the initial HSC gen3 repo was set up, this was worked out rather early on via the `refcats` collection so images could be processed at all. Everything I've done with DECam has been much more ad hoc, because as in many situations, certain refcat defaults were hardcoded into obs_subaru and never thought about for obs_decam.

I guess for now, I will make a `ApPipe_hits2015-3` with these configs that imports all the rest of everything to go with my repo by the same name. And later we can add one for `/repo/main` which will be fun to come up with a name for!

Reply:
All I can find is https://github.com/lsst/obs_subaru/blob/master/config/calibrate.py, which sets `ps1_pv3_3pi_20170110` explicitly. 😕

Reply:
Perhaps they just stick with that name instead of calling it a more succinct "panstarrs." We may have to follow suit for DECam in the shared repo, time will tell.